vllm/model_executor/layers/spec_decode_base_sampler.py

```diff
@@ -77,7 +77,8 @@ def _create_output(
             tensor is [batch_size, k + num_bonus_tokens]
         """
         batch_size, k = substitute_token_ids.shape
-        bonus_token_ids = bonus_token_ids.squeeze()
+        # TODO: HPU has an issue when handling squeeze with all dim as 1
+        bonus_token_ids = bonus_token_ids.squeeze(-1)
```

Comment:
shouldn't it be under if hpu clause?

Reply:
> shouldn't it be under if hpu clause?

@michalkuligowski , I am not getting this question. This is what I observed during debug, do you want me to remove the comments?

Reply:
remove the comments